src/library/psi_library.jl

```diff
@@ -1149,3 +1149,415 @@ function build_two_zone_5_bus(; kwargs...)
+        PSY.add_component!(main_sys, srvc)
+        # check if it has timeseries
+        if PSY.has_time_series(main_comp)
+            PSY.copy_time_series!(srvc, main_comp)
+        end
+    end
+
+    # finally add the remaining devices (lines are not present since removed before)
+    for b in PSY.get_components(Device, twin_sys)
+        name_ = PSY.get_name(b)
+        main_comp = PSY.get_component(typeof(b), main_sys, name_)
+        # remove the component and services
+        PSY.clear_services!(b)
+        PSY.remove_component!(twin_sys, b)
+        # change name
+        PSY.set_name!(b, name_ * "_twin")
+        # change bus (already changed)
+        # check if it has services
+        @assert !PSY.has_service(b, PSY.VariableReserve)
+        #check if component has time_series
+        if !PSY.has_time_series(b)
+            # define time series container
+            IS.assign_new_uuid!(b)
+            # add comopnent to the new sys (main)
+            PSY.add_component!(main_sys, b)
+            PSY.copy_time_series!(b, main_comp)
+        else
+            IS.assign_new_uuid!(b)
+            PSY.add_component!(main_sys, b)
+        end
+        # add service to the device to be added to main_sys
+        if length(PSY.get_services(main_comp)) > 0
+            PSY.get_name(b)
+            srvc_ = PSY.get_services(main_comp)
+            for ss in srvc_
+                srvc_type = typeof(ss)
+                srvc_name = PSY.get_name(ss)
+                PSY.add_service!(
+                    b,
+                    PSY.get_component(srvc_type, main_sys, srvc_name * "_twin"),
+                    main_sys,
+                )
+            end
+        end
+        # change scale
+        if typeof(b) <: RenewableGen
+            PSY.set_base_power!(b, 1.2 * PSY.get_base_power(b))
+            PSY.set_base_power!(main_comp, 0.9 * PSY.get_base_power(b))
+        end
+        if typeof(b) <: PowerLoad
+            PSY.set_base_power!(main_comp, 1.2 * PSY.get_base_power(b))
+        end
+    end
+
+    # conncect two buses: one with a AC line and one with a HVDC line.
+    # Consider area 1 and area 1_twin
+
+    # now look at all the buses in area 1
+    area_ = PSY.get_component(PSY.Area, main_sys, "1")
+    buses_ =
+        [b for b in PSY.get_components(PSY.ACBus, main_sys) if PSY.get_area(b) == area_]
+
+    # get lines for those buses
+    br_in_area = []
+    br_per_bus = Dict(PSY.get_name(b) => [] for b in buses_)
+    br_other_areas = []
+
+    for br in PSY.get_components(PSY.Line, main_sys)
+        if PSY.get_from_bus(br) in buses_ || PSY.get_to_bus(br) in buses_
+            if !(br in br_in_area)
+                push!(br_in_area, br)
+            end
+            if PSY.get_from_bus(br) in buses_
+                if !(PSY.get_name(br) in br_per_bus[PSY.get_name(PSY.get_from_bus(br))])
+                    push!(br_per_bus[PSY.get_name(PSY.get_from_bus(br))], PSY.get_name(br))
```

Comment:
`br_per_bus` also doesn't seem to be used. But if you do need end up needing these, consider making them `Set`s or `OrderedSet`s to avoid having to manually check if they already contain the thing (also technically improves the time complexity, though I doubt it matters)

Reply:
@alefcastelli same here